### PR TITLE
Issue #1686 - jsx-handler-names: false positive when handler name begins with number

### PR DESCRIPTION
**Pull request by jsphstls**
Updated unit test and EVENT_HANDLER_REGEX regex.

### CODE REVIEW

**Comment by jsphstls** [comment]
I adjusted the regex so that a capital letter is still required at some point after the word "handle" and added more test cases to demonstrate that minimum requirement.

File: tests/lib/rules/jsx-handler-names.js [hunk 1]
```diff
@@ -69,6 +73,18 @@ ruleTester.run('jsx-handler-names', rule, {
...
+    errors: [{message: 'Handler function for onChange prop key must begin with \'handle\''}]
+  }, {
+    code: '<TestComponent onChange={this.handle2} />',
+    errors: [{message: 'Handler function for onChange prop key must begin with \'handle\''}]
+  }, {
+    code: '<TestComponent onChange={this.handl3Change} />',
+    errors: [{message: 'Handler function for onChange prop key must begin with \'handle\''}]
+  }, {
+    code: '<TestComponent onChange={this.handle4change} />',
+    errors: [{message: 'Handler function for onChange prop key must begin with \'handle\''}]
```
Comment (ljharb) [inline tests/lib/rules/jsx-handler-names.js]: hm, we should probably change the error message to add " and be in camelCase", otherwise this one's pretty confusing

Comment (jsphstls) [reply tests/lib/rules/jsx-handler-names.js]: I updated the error message to be more specific and removed the handleC test case that does pass, but should not be considered valid.

Comment (ljharb) [reply tests/lib/rules/jsx-handler-names.js]: if the test wrongly passes, we should move it to the "invalid" section, even if it's commented out for now.
